### PR TITLE
Add ApiKey expiration time to audit log

### PR DESCRIPTION
**Pull request by jfreden**
Follow up to PR: https://github.com/elastic/elasticsearch/pull/103453

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java [hunk 1]
```diff
@@ -1325,6 +1325,9 @@ private static void withBaseUpdateApiKeyFields(final XContentBuilder builder, fi
                 // because it replaces any metadata previously associated with the API key
                 builder.field("metadata", baseUpdateApiKeyRequest.getMetadata());
             }
+            if (baseUpdateApiKeyRequest.getExpiration() != null) {
+                builder.field("expiration", baseUpdateApiKeyRequest.getExpiration());
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java:1329]: Not a strong preference, but I'd stay consistent with the create API key event log, that is explicitly log `expiration` as `null`, i.e. sth like: `field("expiration", expiration != null ? expiration.toString() : null)`

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrailTests.java [hunk 2]
```diff
@@ -627,21 +627,23 @@ public void testSecurityConfigChangeEventFormattingForRoles() throws IOException
         CapturingLogger.output(logger.getName(), Level.INFO).clear();
 
         final String keyId = randomAlphaOfLength(10);
+        final TimeValue newExpiration = ApiKeyTests.randomFutureExpirationTime();
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrailTests.java:630]: Nit: I'd randomize between an actual value and `null`, i.e., `final TimeValue newExpiration = randomFrom(ApiKeyTests.randomFutureExpirationTime(), null);` and tweak the assertion further down.

**Review by n1v0lg — APPROVED** [review]
LGTM, sorry for the delay! 

For this one, we can go with a `Security/Audit` label. Also, two follow ups, one related to audit logging, other generically to the addition of the new field (forgot about this one before):

* We should update audit event docs: https://www.elastic.co/guide/en/elasticsearch/reference/current/audit-event-types.html (under `change_apikey` and other event types where it matters) to reflect the new field
* We might need to update the API spec -- this is used to generate ES client definitions. See this PR for an example: https://github.com/elastic/elasticsearch-specification/pull/2371 
*

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java (hunk 1 above)
Comment (jfreden) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java]: Good point! Changed it.

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrailTests.java (hunk 2 above)
Comment (jfreden) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrailTests.java]: Yes, I agree.

**Comment by jfreden** [comment]
Thanks for the review! 

> We should update audit event docs: https://www.elastic.co/guide/en/elasticsearch/reference/current/audit-event-types.html (under change_apikey and other event types where it matters) to reflect the new field

Add it to this PR. 

> We might need to update the API spec -- this is used to generate ES client definitions. See this PR for an example: https://github.com/elastic/elasticsearch-specification/pull/2371

Will work on a separate PR for this.

**Pull request merged**